web_poet/fields.py

```diff
@@ -63,6 +73,28 @@ def __init__(self, method):
                 raise TypeError(
                     f"@field decorator must be used on methods, {method!r} is decorated instead"
                 )
+            if out:
+                orig_method = method
```

Comment:
This is needed because, as I've just found out, Python closures get the final value of the variable, not the one at the point where the closure is defined, so `method = processed` is applied when the closure is called.